cime_config/tests.py

```diff
@@ -346,13 +344,12 @@
     "e3sm_prod" : {
         "inherit" : "e3sm_atm_prod",
         "tests"   : (
-            "SMS_Ld1.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.allactive-wcprod_1850_r05",
```

Comment:
Duplicate of `SMS_Ld1.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.allactive-wcprod_1850`. Also removed test-mod dir `allactive-wcprod_1850_r05`, which is no longer used by anything else.